Sources/StreamChatUI/Controller/ChatAddFriendVC.swift

```diff
@@ -78,22 +70,53 @@ public class ChatAddFriendVC: ChatBaseVC {
         chatUserList.sortType = .sortByAtoZ
         chatUserList.viewModel.existingUsers = existingUsers
         chatUserList.viewModel.fetchUserList()
+        // Add friend Callback
+        chatUserList.bCallbackAddFriend = { [weak self] user in
+            guard let weakSelf = self else { return }
+            guard let selectedUser = user else { return }
+            weakSelf.searchField.resignFirstResponder()
+            if weakSelf.selectionType == .addFriend {
+                weakSelf.showAddFriendConfirmPopup(user: selectedUser)
+            }
+        }
+        // Setup UI
         viewContainerLeadingConst.constant = 5
         viewContainerTrailingConst.constant = 5
         setupUI()
     }
     
     private func setupUI() {
-        let cornorRadius = viewContainerLeadingConst.constant > 0 ? 32 : 0
-        
+        let cornerRadius = viewContainerLeadingConst.constant > 0 ? 32 : 0
         viewHeaderView.backgroundColor = Appearance.default.colorPalette.chatViewBackground
         searchBarContainerView.backgroundColor = Appearance.default.colorPalette.searchBarBackground
         searchBarContainerView.layer.cornerRadius = 20.0
-        viewHeaderView.layer.cornerRadius = CGFloat(cornorRadius)
+        viewHeaderView.layer.cornerRadius = CGFloat(cornerRadius)
     }
     
     @objc private func textDidChange(_ sender: UITextField) {
-        self.chatUserList.viewModel.searchDataUsing(searchString: sender.text)
+        if let searchText = sender.text, searchText.isEmpty == false {
+            if searchText.containsEmoji || searchText.isBlank {
+                return
+            }
+            self.chatUserList.viewModel.searchDataUsing(searchString: searchText)
+        } else {
+            self.chatUserList.viewModel.searchText = nil
```

Comment:
@jitendra-simform  Please remove self. It is not required here.

Reply:
Using self here does not create any issue , could you please share any specific reason to remove self from here

Reply:
@jitendra-simform  Please let me know any specific reason for using 'self'? 
Does class function/variable does not access without 'self'?

Added this comment because multiple developers are working in the same repo. We must need to maintain code consistency.

Reply:
@ajay-simform  Fixed , Please check